File: stara/2005/python/zad5_a.py
# dane podane w zadaniu
data = [1, -2, 6, -5, 7, -3]

# najlepsza suma
best_sum = -float('inf')
# przechodze po kazdym zakresie
for start in range(len(data)):
    for end in range(start + 1, len(data) + 1):
        # zliczenie sumy
        curr = sum(data[start:end])
        # jezeli suma wieksza od rekordu to zapisz
        if curr > best_sum:
            best_sum = curr


# wyswietlenie odpowiedzi
answer = f'5 a) Najlepsza suma: {best_sum}'
print(answer)

File: stara/2005/c++/zad5_a.cpp
#include <iostream>
#include <vector>
using namespace std;

string zad5_a()
{
    vector<int> data = {1, -2, 6, -5, 7, -3};
    int max = -9999;

    for (int i = 0; i < data.size() - 1; i++)
    {
        int sum = 0;

        for (int j = i; j < data.size() - 1; j++)
        {
            sum += data[j];

            if (sum > max)
                max = sum;
        }
    }

    return "5 a) Najlepsza suma podanego ciagu: " + to_string(max);
}
